src/Features/CSharp/Portable/ConvertLinq/ConvertForEachToLinqQuery/SyntaxNodeAndSyntaxTokenExtensions.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.ConvertLinq.ConvertForEachToLinqQuery;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertLinq.ConvertForEachToLinqQuery
+{
+    internal static class SyntaxNodeAndSyntaxTokenExtensions
+    {
+        public static T WithComments<T>(this T node, IEnumerable<SyntaxToken> leadingTokens, IEnumerable<SyntaxToken> trailingTokens) where T : SyntaxNode
+            => node.WithComments(Helpers.GetTrivia(leadingTokens), Helpers.GetTrivia(trailingTokens));
```

Comment:
WithCommentsFrom would likely be a better name.  Note: we have some helpers for this already.  So these could be siblings of those helpers.